lib/Updater.php

```diff
@@ -596,13 +618,47 @@ public function downloadUpdate(): void {
 			$message .= ' - URL: ' . htmlentities($response['url']);
 
 			throw new \Exception($message);
+		} else {
+			// download succeeded
+			$info = curl_getinfo($ch);
+			$this->silentLog("[info] download stats: size=" . $this->formatBytes($info['size_download']) . " bytes; total_time=" . round($info['total_time'], 2) . " secs; avg speed=" . $this->formatBytes($info['speed_download']) . "/sec");
 		}
+
 		curl_close($ch);
 		fclose($fp);
 
 		$this->silentLog('[info] end of downloadUpdate()');
 	}
 
+	private function downloadProgressCallback(\CurlHandle $resource, int $download_size, int $downloaded, int $upload_size, int $uploaded): void {
+		static $previousProgress = 0;
```

Comment:
Why static? It should be a property set to 0 before starting the download I think.

Reply:
Because that function is invoked throughout the download and that variable tracks whether there has been any progress since the last invocation.

Reply:
Fixed. Non-OOP brain still learning new habits. :-P